### PR TITLE
Run tests with new resolve on CI

### PR DESCRIPTION
**Pull request by dima74**
Add another job into `check` workflow to run tests with [new resolve](https://github.com/intellij-rust/intellij-rust/issues/6217) enabled.

Also add annotation `@IgnoreInNewResolve` to temporary disable few tests if running with new resolve.

### CODE REVIEW

**Review by vlad20012 — APPROVED** [review]

File: src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ import org.rust.openapiext.toPsiFile
 
 val isNewResolveEnabled: Boolean
     get() = isFeatureEnabled(RsExperiments.RESOLVE_NEW_ENGINE)
+        || System.getenv("INTELLIJ_RUST_FORCE_USE_NEW_RESOLVE") != null
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt:30]: Maybe `isUnitTestMode && ...`

**Review by vlad20012 — APPROVED** [review]
:+1: 
Please, rebase to the latest master

**Comment by dima74** [comment]
bors r=vlad20012

**Comment by dima74** [comment]
bors r=vlad20012

**Pull request merged**